framework/src/NormalisedSumPDF.cpp

```diff
@@ -248,12 +245,24 @@ void NormalisedSumPDF::MakePrototypes( PhaseSpaceBoundary * InputBoundary )
 			}
 		}
 	}
+	// Make sure the cached integrals have the right indices. Set everything to 1 to avoid division by zero. Old cache is invalid anyway.
```

Comment:
`MakePrototypes` is called upon construction of the PDF and also when a new `PhaseSpaceBoundary` is set. I don't know whether the new boundary has the same set of discrete indices as the old one, so the `std::map<unsigned, double>` object that stores the integrals might return 0 if the likelihood is evaluated before the integral, leading to division by zero errors.